### ISSUE TITLE
Hierarchical test DataLoader dict with debug dataset

### ISSUE THREAD

**Issue by petteriTeikari** [minivess-mlops/minivess-mlops#90]
## Summary
Create a two-level hierarchical DataLoader dictionary for multi-dataset generalization evaluation:

```python
dataloaders_dict = {
    "minivess_debug": {
        "all": DataLoader(...)  # 4 random volumes from minivess
    },
    "ucl_2pm": {
        "all": DataLoader(...)
    },
    "harvard_3pm": {
        "thin_vessels": DataLoader(...),
        "thick_vessels": DataLoader(...),
        "mixed": DataLoader(...),
        "all": DataLoader(...)
    }
}
```

Also support hierarchical validation DataLoaders from training folds.

### Debug Dataset
Create n=4 random synthetic volumes (or sampled from minivess) as a debug/dev test set for mechanics testing while waiting for real external test datasets.

## Files
- **New:** `src/minivess/data/test_datasets.py` — DatasetRegistry, build_hierarchical_dataloaders()
- **New:** `src/minivess/data/debug_dataset.py` — create_debug_dataset() (n=4 synthetic volumes)
- **Test:** `tests/v2/unit/test_test_datasets.py`
- **Test:** `tests/v2/unit/test_debug_dataset.py`

## Acceptance Criteria
- [ ] `DatasetEntry` pydantic model with `name`, `subsets`, `data_dir`, `layout`
- [ ] `DatasetRegistry` manages multiple datasets with YAML config
- [ ] `build_hierarchical_dataloaders()` returns nested dict of DataLoaders
- [ ] `create_debug_dataset()` generates 4 synthetic NIfTI volumes with labels
- [ ] Subsets like "thin_vessels" are metadata-filtered views of the parent dataset
- [ ] ~12 tests

## Design Notes
- Each external dataset gets its own top-level key → separate evaluation per dataset
- Subsets allow per-class or per-characteristic evaluation (e.g., thin vs thick vessels)
- The same pattern works for validation dataloaders during training

## Plan Phase
Phase 4 of `docs/planning/evaluation-and-ensemble-execution-plan.xml`

**Comment by petteriTeikari** [comment]
Implemented in 1a99b5e: Hierarchical DataLoader dict + debug dataset generator. 14 tests.
